info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartsetFigure.java

```diff
@@ -26,9 +26,9 @@ public ChartsetFigure(ActionListener addChartHandler)
     chartsetHeader.setTextAlignment(PositionConstants.TOP);
     if (boldFont == null)
     {
```

Comment:
@IanMayo Even with this change (default display) I'm still getting a NPE. Here's the scenario: I'm launching the Limpet run configuration from the IDE, the splash screen appears and then I immediately switch back to the IDE (taking the focus from the launching Limpet). That casues the Display.getDefault().getActiveShell() to be null and results in 
```
java.lang.NullPointerException
	at info.limpet.stackedcharts.ui.editor.figures.ChartsetFigure.<init>(ChartsetFigure.java:29)
	at info.limpet.stackedcharts.ui.editor.parts.ChartSetHeaderEditPart.createFigure(ChartSetHeaderEditPart.java:49)
	at org.eclipse.gef.editparts.AbstractGraphicalEditPart.getFigure(AbstractGraphicalEditPart.java:494)
	at org.eclipse.gef.editparts.AbstractGraphicalEditPart.addChildVisual(AbstractGraphicalEditPart.java:208)
	at org.eclipse.gef.editparts.AbstractEditPart.addChild(AbstractEditPart.java:211)
....
```
I assume you should be able to reproduce that on your end as well. Let me know if that's not the case. My proposal is to initialize the boldFont a bit later, when the graphic context is somehow ready. I'll push a proposed fix bit later. Normally Display.getCurrent() will return null when invoked from a non-UI thread and Display.getDefault() will always return the Display instance for the SWT UI thread.